.github/workflows/ngen_integration.yaml

```diff
@@ -50,6 +50,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           repository: noaa-owp/soilmoistureprofiles
+          ref: refs/pull/17/head
```

Comment:
Why is this checking out a PR ref? Does that PR need to merge then we checkout HEAD or tag a commit to reference here? I didn't think we want the CI checking out a constant PR ref, unless I'm missing something here